### PR TITLE
PR to fix a few syntax and typo errors

### PR DESCRIPTION
**Pull request by HemanthD**
This PR fixes the syntax warning/error due to unwanted extra ; in a few templates reported in #3800. The PR also includes fixes to a few typo errors and also formatting corrections (removing extra blank lines, adjusting format).

### CODE REVIEW

File: generators/server/templates/src/main/java/package/domain/_User.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 import java.time.ZonedDateTime;<% } %>
 
 /**
- * A user.
+ * An user.
```
Comment (ruddell) [inline generators/server/templates/src/main/java/package/domain/_User.java]: `An` should be `A` here.  

When `u` makes the same sound as the `y` in `you`, or `o` makes the same sound as `w` in `won`, then `a` is used.

File: generators/server/templates/src/main/java/package/security/_UserDetailsService.java [hunk 2]
```diff
@@ -16,7 +16,7 @@
 import java.util.stream.Collectors;
 
 /**
- * Authenticate a user from the database.
+ * Authenticate an user from the database.
```
Comment (ruddell) [inline generators/server/templates/src/main/java/package/security/_UserDetailsService.java]: `an` should be `a` here.

File: generators/server/templates/src/main/java/package/domain/_User.java (hunk 1 above)
Comment (HemanthD) [inline generators/server/templates/src/main/java/package/domain/_User.java]: Thanks for the review. I have reverted the changes from 'An user' to 'A user'

**Comment by deepu105** [comment]
thanks and appreciated :+1:

**Pull request merged**